### PR TITLE
Sebastianmaj/gc vars

### PR DESCRIPTION
**Pull request by SebastianMaj**
**Issue**

- Link: https://github.com/NebraLtd/hm-miner/issues/62
- Summary: Add storage limit for GC activation and limit the amount of latest blocks stored on the miner

**How**
With firmware 2021.12.29.2 helium introduced 2 new variables in blockchain-core, these consist of ```BLOCKCHAIN_ROCKSDB_GC_BYTES``` in the environment file which is a bytes value for the limit on local storage of block data before GC is activated and  ```blocks_to_protect_from_gc``` which is stored in sys.config and it tells the miner what's the limit on the number of recent blocks stored.

**References**
[](https://engineering.helium.com/2021/12/30/blockchain-release-snapshot-loading-and-storage-improvements.html)

**Checklist**

- [x] Cleaned up commit history (rebase!)
- [ ] Documentation added
- [ ] Thought about variable and method names

### CODE REVIEW

File: start-miner.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env sh
 
 . ./dbus-wait.sh
+. ./miner-gc.sh
```
Comment (vpetersson) [inline start-miner.sh:4]: No, this is wrong. There's no need to do this. We either do it in `docker-compose.yml` or we do it in the [Dockerfile](https://github.com/NebraLtd/hm-miner/blob/master/Dockerfile). No point in reinventing the wheel here.

**Review by vpetersson — CHANGES_REQUESTED** [review]

**Comment by shawaj** [comment]
will need to do corresponding updates in https://github.com/NebraLtd/hm-block-tracker as well

Specifically in: https://github.com/NebraLtd/hm-block-tracker/blob/master/config.template

**Comment by shawaj** [comment]
Also should add some README information about the params and why we chose those values etc etc

**Comment by KevinWassermann94** [comment]
The testnet is currently running 8GB and 10000 blocks to be protected. The storage usage seems to be maxing out around 17-18.5GB of storage usage with the garbage collection working 1-2 times per day cleaning around 1GB.
The blockchain size still seems to be around 14GB, which leads to the conclusion that the 10000 blocks are the limiting factor.

I would like to reduce the protected blocks to 4000 to improve the storage usage and therefore the performance.

**Comment by shawaj** [comment]
4000 seems reasonable, pending testing

**Comment by shawaj** [comment]
PR in block tracker https://github.com/NebraLtd/hm-block-tracker/pull/60

File: docker.config.rockpi [hunk 2]
```diff
@@ -7,7 +7,8 @@
     ]},
     {blockchain,
     [
-      {key, {ecc, [{key_slot, 0}, {onboarding_key_slot, 0}, {bus, "i2c-7"}]}}
+      {key, {ecc, [{key_slot, 0}, {onboarding_key_slot, 0}, {bus, "i2c-7"}]}},
+      {blocks_to_protect_from_gc, 5000}
```
Comment (shawaj) [inline docker.config.rockpi:11]: These are set to 5000 currently so maybe just change to 4000 as per @KevinWassermann94 and then looks good to go to me for testing

Comment (SebastianMaj) [reply docker.config.rockpi]: Perfect, changes have been applied. I set it to 4,000 in all 3 configs

**Review by shawaj — APPROVED** [review]

**Comment by shawaj** [comment]
@SebastianMaj can you merge the block tracker one at same time?

**Comment by SebastianMaj** [comment]
@shawaj Thank you for merging both

**Comment by shawaj** [comment]
No probs! Kevin wanted to get it in so just did it 😊

**Pull request merged**